WordPress/src/main/java/org/wordpress/android/ui/stories/StoryMediaSaveUploadBridge.kt

```diff
@@ -0,0 +1,170 @@
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.ui.posts.EditPostActivity.OnPostUpdatedFromUIListener
+import org.wordpress.android.ui.posts.EditPostRepository
+import org.wordpress.android.ui.posts.PostUtils.WP_STORIES_POST_MEDIA_LOCAL_ID_PLACEHOLDER
+import org.wordpress.android.ui.posts.PostUtilsWrapper
+import org.wordpress.android.ui.posts.SavePostToDbUseCase
+import org.wordpress.android.ui.posts.editor.media.AddLocalMediaToPostUseCase
+import org.wordpress.android.ui.posts.editor.media.EditorMediaListener
+import org.wordpress.android.ui.uploads.UploadServiceFacade
+import org.wordpress.android.util.NetworkUtilsWrapper
+import org.wordpress.android.util.helpers.MediaFile
+import javax.inject.Inject
+import javax.inject.Named
+import kotlin.coroutines.CoroutineContext
+
+/*
+ * StoryMediaSaveUploadBridge listens for StorySaveResult events triggered from the StorySaveService, and
+ * then transforms its result data into something the UploadService can use to upload the Story frame media
+ * first, then obtain the media Ids and collect them, and finally create a Post with the Story block (
+ * (simple WP gallery for alpha) with he obtained media Ids.
+ * This is different than uploading media to a regular Post because we don't need to replace the URLs for final Urls as
+ * we do in Aztec / Gutenberg.
+ * The gallery is only a collection of media Ids, so we really need to first upload the Media, obtain the remote id,
+ * and then only create the post and upload it.
+ */
+class StoryMediaSaveUploadBridge @Inject constructor(
+    private val addLocalMediaToPostUseCase: AddLocalMediaToPostUseCase,
+    private val savePostToDbUseCase: SavePostToDbUseCase,
+    private val uploadService: UploadServiceFacade,
+    private val networkUtils: NetworkUtilsWrapper,
+    private val postUtils: PostUtilsWrapper,
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
+) : CoroutineScope, LifecycleObserver, EditorMediaListener {
+    // region Fields
+    private var job: Job = Job()
+
+    override val coroutineContext: CoroutineContext
+        get() = mainDispatcher + job
+
+    @Inject lateinit var editPostRepository: EditPostRepository
+
+    @Suppress("unused")
+    @OnLifecycleEvent(ON_CREATE)
+    fun onCreate(source: LifecycleOwner) {
+        EventBus.getDefault().register(this)
+    }
+
+    @Suppress("unused")
+    @OnLifecycleEvent(ON_DESTROY)
+    fun onDestroy(source: LifecycleOwner) {
+        // note: not sure whether this is ever going to get called if we attach it to the lifecycle of the Application
+        // class, but leaving it here prepared for the case when this class is attached to some other LifeCycleOwner
+        // other than the Application.
+        cancelAddMediaToEditorActions()
+        EventBus.getDefault().unregister(this)
+    }
```

Comment:
Could we use the `eventBusWrapper` here since we are going to be testing this class in the near future? WDYT?

Reply:
👍 made the change in 86fb2ad01d - also adding the `removeStickyEvent()` method to the wrapper